src/kernel.h

```diff
@@ -30,20 +31,24 @@ class quantum_kernel {
     utils::UInt             iterations;
     utils::UInt             qubit_count;
     utils::UInt             creg_count;
+    utils::UInt             breg_count;
     kernel_type_t           type;
     circuit                 c;
     utils::Bool             cycles_valid; // used in bundler to check if kernel has been scheduled
     utils::Opt<operation>   br_condition;
     utils::UInt             cycle_time;   // FIXME HvS just a copy of platform.cycle_time
     instruction_map_t       instruction_map;
+    utils::Vec<utils::UInt> cond_operands;    // see gate interface: condition mode to make new gates conditional
```

Comment:
parameters of quantum_kernel::gate() were added at the end to maintain backward compatibility, but internally I would maintain a more natural order.

Reply:
Yes, true, will have a look.
Although with cleaning up the IR, this will change anyhow.

Reply:
Propose to leave them there.
These are not the arguments of a gate (see src/gate.h, there I rearranged the fields) but
meant to keep the preset_condition (see the related interface).